store/src/java/com/zimbra/cs/filter/ZimbraMailAdapter.java

```diff
@@ -529,8 +565,16 @@ public Message keep(KeepType type) throws ServiceException {
         String folderPath = handler.getDefaultFolderPath();
         folderPath = CharMatcher.is('/').trimFrom(folderPath); // trim leading and trailing '/'
         Message msg = null;
-        ZimbraLog.filter.debug(type == KeepType.EXPLICIT_KEEP ? "Explicit - fileinto " : "Implicit - fileinto " +
-            appendFlagTagActionsInfo(folderPath, getFlagActions(), getTagActions()));
+        if (LC.lmtp_extended_logs_enabled.booleanValue()) {
+            String messageId = Mime.getMessageID(handler.getMimeMessage());
+            String sender = Mime.getSender(handler.getMimeMessage());
+            ZimbraLog.filter.info(type == KeepType.EXPLICIT_KEEP ? "Explicit - fileinto " : "Implicit - fileinto " +
```

Comment:
I would not write these conditions in a log statement, better to separate out that. It's hard to understand

Reply:
done